app/uk/gov/hmrc/cataloguefrontend/auth/CatalogueAuthBuilders.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.auth
+
+// import play.api.mvc._
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.internalauth.client.FrontendAuthComponents
+import play.api.mvc.{ActionBuilder, AnyContent, BodyParser, Call, MessagesControllerComponents, MessagesRequest, Result, Request}
+
+trait CatalogueAuthBuilders {
+
+  def auth: FrontendAuthComponents
+  def ec: ExecutionContext
+  def mcc: MessagesControllerComponents
+
+  def BasicAuthAction =
+    new ActionBuilder[MessagesRequest[*], AnyContent] {
+      override def executionContext: ExecutionContext = ec
+      override def parser: BodyParser[AnyContent] = mcc.parsers.anyContent
```

Comment:
Interesting that internal-auth is hard-coded to `AnyContent` wonder if we can parametrise this to anything we can provide an implicit BodyParser.

Reply:
This is fine - this is the way `Action` behaves too. You can provide an alternative `BodyParser` to the `apply` and `async` functions.